### PR TITLE
generate new `Sender` if hd_index was provided in set_options

### PR DESCRIPTION
**Pull request by Buckram123**
hd_index impacts private key meaning if it's provided inside `SenderOptions` new Sender should be created from scratch
Closes ORC-127, Closes ORC-125

### CODE REVIEW

File: cw-orch-daemon/src/sync/builder.rs [hunk 1]
```diff
@@ -227,4 +228,23 @@ mod test {
...
+    #[test]
+    #[serial_test::serial]
+    fn hd_index_re_generates_sender() -> anyhow::Result<()> {
+        let daemon = DaemonBuilder::default()
+            .chain(OSMOSIS_1)
+            .mnemonic(DUMMY_MNEMONIC)
+            .build()
+            .unwrap();
+
+        let indexed_daemon = daemon.rebuild().hd_index(56).build().unwrap();
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply cw-orch-daemon/src/sync/builder.rs:241]: Yes, that's what this PR tries to fix

**Review by Kayanski — APPROVED** [review]
Very nice ! 
We will need to support updating the daemon state as well when rebuilding, but that's another task.

File: cw-orch-daemon/src/sync/builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (CyberHoward) [reply cw-orch-daemon/src/sync/builder.rs:241]: Then why are you asserting that the address basically remains the same? This test confuses me a bit

**Comment by CyberHoward** [comment]
> Very nice ! We will need to support updating the daemon state as well when rebuilding, but that's another task.

What do you mean by this?

File: cw-orch-daemon/src/sync/builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Buckram123) [reply cw-orch-daemon/src/sync/builder.rs:241]: `assert_ne` is not equal, before this PR they `eq`, and now they `ne`

**Pull request merged**